modules/seqexec/server/src/main/scala/seqexec/server/gsaoi/GsaoiEpics.scala

```diff
@@ -0,0 +1,182 @@
+
+  def simulationMode: F[Option[String]] = safeAttribute(status.getStringAttribute("DCSIM"))
+
+  def timingBoardCodeName: F[Option[String]] = safeAttribute(status.getStringAttribute("DSPTIMBN"))
+
+  def readInterval: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("READDLAY"))
+
+  def detectorTemperature: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("DETTEMP"))
+
+  def upperFilterHealth: F[Option[String]] = safeAttribute(status.getStringAttribute("FILT1CAR"))
+
+  def dewarPressure: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("DEWPRES"))
+
+  def obsElapsedTime: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("ELAPSED"))
+
+  def lowerFilterEngPos: F[Option[Int]] = safeAttributeSInt(status.getIntegerAttribute("FILT2POS"))
+
+  def resetDelay: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("RSTDLAY"))
+
+  def detectorHousingTemperature: F[Option[Double]] =
+    safeAttributeSDouble(status.getDoubleAttribute("DETHTEMP"))
+
+  def utilityWheelEngPos: F[Option[Int]] = safeAttributeSInt(status.getIntegerAttribute("UTLWPOS"))
+
+  def utilityWheelHealth: F[Option[String]] = safeAttribute(status.getStringAttribute("UTLWCAR"))
+
+  def windowCoverHealth: F[Option[String]] = safeAttribute(status.getStringAttribute("CVERCAR"))
+
+  def upperFilterEngPos: F[Option[Int]] = safeAttributeSInt(status.getIntegerAttribute("FILT1POS"))
+
+  def expositionMode: F[Option[String]] = safeAttribute(status.getStringAttribute("EXPMODE"))
+
+  def dcName: F[Option[String]] = safeAttribute(status.getStringAttribute("DCNAME"))
+
+  def pciBoardCodeName: F[Option[String]] = safeAttribute(status.getStringAttribute("DSPPCIN"))
+
+  def readTime: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("READTIME"))
+
+  def requestedExposureTime: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("exposureTime"))
+
+  def numberOfResets: F[Option[Int]] = safeAttributeSInt(status.getIntegerAttribute("numberOfResets"))
+
+  def readMode: F[Option[String]] = safeAttribute(status.getStringAttribute("readMode"))
+
+  def numberOfFowlerSamples: F[Option[Int]] = safeAttributeSInt(status.getIntegerAttribute("numberOfFowSamples"))
+
+  def coadds: F[Option[Int]] = safeAttributeSInt(status.getIntegerAttribute("coadds"))
+
+  def exposedTime: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("exposedTime"))
+
+  def timeMode: F[Option[String]] = safeAttribute(status.getStringAttribute("timeMode"))
+
+  def countdown: F[Option[Double]] =  safeAttributeSDouble(status.getDoubleAttribute("countdown"))
+
+  private val dhsConnectedAttr: CaAttribute[DhsConnected] =
+    status.addEnum[DhsConnected]("dhsConnected", s"${GsaoiTop}sad:dc:dhsConnO", classOf[DhsConnected])
+  def dhsConnected: F[Option[DhsConnected]] = safeAttribute(dhsConnectedAttr)
+
+
+}
+
+object GsaoiEpics extends EpicsSystem[GsaoiEpics[IO]] {
```

Comment:
I kind of want to migrate away from the `XXXEpics` objects and de them as classes parametrized on `F[_]` Do you think we could do that here?

Reply:
`GsaoiEpics` is already a class paramtrized on `F[_]`. You are looking here at the companion object.